### ISSUE TITLE
DAR402 wrongly reported on bare raise

### ISSUE THREAD

**Issue by fredrikaverpil** [terrencepreilly/darglint#134]
With this code, I get `Excess exception(s) in Raises section: +r TypeErrorflake8(DAR402)` with darglint 1.5.5 via flake8:

```python
"""Test."""

from somewhere import y, z


def x():
    """Do stuff.

    Raises:
        TypeError: wrong type
    """
    try:
        y()
    except TypeError as err:
        z(err)
        raise
```

I feel this is wrong, as I am inside of `x()` indeed re-raising `TypeError`.

Related:
- https://github.com/terrencepreilly/darglint/issues/102
- https://github.com/terrencepreilly/darglint/commit/4c615c9d054f6f17ba75f210fc72c0e7490b7d9e

**Comment by fredrikaverpil** [comment]
@terrencepreilly hi, I was just wondering if you agree with me that the above reported issue - or if this is a misunderstanding on my part.

**Comment by terrencepreilly** [comment]
Yes, confirmed that this is handled incorrectly.  I just haven't had time to handle this yet.  I'll give it a look this weekend.

**Comment by terrencepreilly** [comment]
This should be resolved in b33b261e43bd31383efe71f7b9295cdf58c9857b, and the fix will be present in v1.5.6, which I'll try to push this weekend.

**Comment by terrencepreilly** [comment]
It should now be available.

**Comment by fredrikaverpil** [comment]
Hi @terrencepreilly I am noticing that with v1.5.7, a bare return now requires a "Returns:" statement in the docstring. Was this a side-effect of allowing bare raising?

If a function never returns anything (always returns None) but you want the function to exit early, it's common practice to do a "bare return". Would you say the intent here is to require a "Returns:" statement in the dosctring in such cases?

**Comment by fredrikaverpil** [comment]
Ah, I just saw [your comment](https://github.com/terrencepreilly/darglint/issues/141#issuecomment-732544157) on that this should be fixed in 1.5.8 👍

**Comment by fredrikaverpil** [comment]
Hi again @terrencepreilly 

What about this scenario?
With darglint 1.5.8, I still get "excessive exceptions in Raises section" (DAR402) for both HTTPError and ConnectionError

```python
"""ddd."""

import requests


def something():
    """Something."""
    pass


def test_something():
    """Test.

    Raises:
        HTTPError: if call to something fails...
        ConnectionError: if call to something fails...

    """
    try:
        something()
    except (requests.HTTPError, requests.ConnectionError) as err:
        print(err)
        raise
```

**Comment by stephen-dexda** [comment]
Similarly, as of 1.5.8:

```python
def foo():
    """Try bar() 5 times.

    Raises:
        Exception: if bar() excepts 5 times.

    """
    for _try in range(5):
        try:
            bar()
        except Exception as exc:
            last_exc = exc
    raise last_exc
```

```bash
DAR401: Missing exception(s) in Raises section: -r last_exc
DAR402: Excess exception(s) in Raises section: +r Exception
```

**Comment by billsioros** [comment]
Similarly, as of 1.8.0:

```python
@classmethod
    def search(cls, query: str, limit: int = 1) -> Iterator["Model"]:
        """Perform a Spotify search given a `query`.

        Args:
            query (str): the search `query`
            limit (int, optional): the number of items to return. Defaults to 1.

        Raises:
            cls.NotFound: In case no results corresponding to the provided query
            are found

        Returns:
            Iterator["Model"]: the `Model` instances corresponding to the query
        """
        view_name = cls.view_name()

        results = cls.context.search(view_name, query, limit=limit)

        if not results:
            raise cls.NotFound

        return map(lambda kwargs: cls(**kwargs), results)
```

```bash
```
